Integrate OpenMM minimization tool into DeepCritical

EmployeeNo427 opened the pull request:
## Summary
- add an OpenMM-based minimization and short dynamics tool with registry integration
- cover the new OpenMM tool with a lightweight water PDB test
- include the OpenMM runtime dependency and refresh the lockfile

## Testing
- pytest tests/ -m "not optional and not containerized"
- ruff format --check DeepResearch/ tests/
- uvx ty check

<!-- greptile_comment -->

<h3>Greptile Summary</h3>

This PR integrates an OpenMM-based energy minimization and short molecular dynamics tool (`OpenMMMinimizationTool`) into the DeepCritical tool registry, adds a corresponding pytest test, and pulls in `openmm>=8.2.0` as a runtime dependency. The tool follows established codebase patterns (`@dataclass` + explicit `__init__`, `ToolRunner`/`ToolSpec`/`ExecutionResult`, and `registry.register()`) and is intentionally lightweight (custom LJ + harmonic bonds + positional restraints, CPU-only) to run in CI.

Two issues worth addressing before merge:

- **Private OpenMM import** (`DeepResearch/src/tools/openmm_tools.py` line 24): `from openmm.openmm import CustomExternalForce, HarmonicBondForce` reaches into the internal C-extension module rather than the public `openmm` top-level namespace. Both classes are available via `from openmm import ...` and that form is stable across versions.
- **Non-physical test geometry** (`tests/test_tools/test_openmm_tools.py` lines 7–13): The water PDB fixture places hydrogen atoms at O–H distances of ~0.095 Å instead of the expected ~0.96 Å — a likely factor-of-10 decimal-point typo. The test only asserts `math.isfinite(minimized_energy)`, which will pass even from an astronomically high-energy starting configuration, so the bug does not cause a failure but means the test does not exercise realistic molecular geometry.
- **Undocumented optional parameters** (`DeepResearch/src/tools/openmm_tools.py` lines 46–62): `simulation_steps`, `temperature_kelvin`, `friction_coeff_ps`, and `timestep_fs` are silently consumed by `run()` but are absent from `ToolSpec.inputs`, making them invisible to registry-level introspection.

<h3>Confidence Score: 3/5</h3>

- Safe to merge after fixing the test geometry and the private import; the core implementation logic is sound.
- The tool logic and registry integration are correct and follow established patterns. However, the test fixture contains a likely decimal-point typo that makes it a non-representative smoke test, and the import of `openmm.openmm` creates a fragile dependency on a private C-extension. These are not blockers but should be addressed before merging to avoid future breakage and to ensure the test actually validates realistic molecular simulation behaviour.
- `tests/test_tools/test_openmm_tools.py` (incorrect water geometry) and `DeepResearch/src/tools/openmm_tools.py` (private import, undocumented optional params)

<details><summary><h3>Important Files Changed</h3></summary>




| Filename | Overview |
|----------|----------|
| DeepResearch/src/tools/openmm_tools.py | New OpenMM energy minimization tool with LJ/harmonic force field; imports from private `openmm.openmm` C-extension rather than the public API, and four optional simulation parameters are undocumented in `ToolSpec.inputs`. |
| tests/test_tools/test_openmm_tools.py | Single test for the new tool; the water PDB fixture has O–H distances of ~0.095 Å (10× too small due to apparent decimal-point typo), resulting in a physically non-representative test that only weakly validates correct behaviour. |
| DeepResearch/src/tools/__init__.py | Adds `openmm_tools` module import and `__all__` entry; also removes some inline section comments — no functional issues. |
| pyproject.toml | Adds `openmm>=8.2.0` as a main runtime dependency; version constraint is appropriate. |
| uv.lock | Lockfile refreshed to add openmm and update existing package entries with upload-time metadata; no concerns. |

</details>


</details>


<h3>Sequence Diagram</h3>

```mermaid
sequenceDiagram
    participant Caller
    participant OpenMMMinimizationTool
    participant ToolRegistry
    participant OpenMM

    Caller->>ToolRegistry: register("openmm_minimize", OpenMMMinimizationTool)
    Caller->>OpenMMMinimizationTool: run({"pdb_contents": ..., "simulation_steps": N})
    OpenMMMinimizationTool->>OpenMMMinimizationTool: validate(params)
    OpenMMMinimizationTool->>OpenMM: PDBFile(StringIO(pdb_data))
    OpenMMMinimizationTool->>OpenMMMinimizationTool: _build_system(pdb)
    Note over OpenMMMinimizationTool: Build System with NonbondedForce,<br/>HarmonicBondForce, CustomExternalForce
    OpenMMMinimizationTool->>OpenMM: Simulation(topology, system, LangevinIntegrator)
    OpenMMMinimizationTool->>OpenMM: context.setPositions(positions)
    OpenMMMinimizationTool->>OpenMM: context.getState(getEnergy=True) → initial_energy
    OpenMMMinimizationTool->>OpenMM: minimizeEnergy(maxIterations=100)
    OpenMMMinimizationTool->>OpenMM: context.getState(getEnergy=True, getPositions=True) → minimized_energy
    OpenMMMinimizationTool->>OpenMM: PDBFile.writeFile(...) → minimized_pdb
    loop N simulation steps
        OpenMMMinimizationTool->>OpenMM: simulation.step(1)
        OpenMMMinimizationTool->>OpenMM: context.getState(getEnergy=True) → trajectory[i]
    end
    OpenMMMinimizationTool-->>Caller: ExecutionResult(energies, trajectory, minimized_pdb)
```

<sub>Last reviewed commit: 66ebc7d</sub>

<!-- greptile_other_comments_section -->

<!-- /greptile_comment -->

EmployeeNo427 committed:
Add OpenMM minimization tool

EmployeeNo427 committed:
Fix lint issues in OpenMM tooling imports

Josephrp commented:
@EmployeeNo427 kindly review the above and push to merge :-) with all my thanks

Josephrp committed:
Merge branch 'dev' into integrate-openmm-into-deepcritical-toolbox

Josephrp reviewed (changes requested):
address the lacunas identified to resolve to the automerge queu :-)

EmployeeNo427 committed:
Fixed OpenMM tool metadata and test realism

-Updated `ToolRunner.validate()` in `base.py` so input descriptors containing `(optional` were not treated as required, while still applying the existing string-type checks against the normalized base type.

-Switched `CustomExternalForce` and `HarmonicBondForce` to import from the public `openmm` namespace in `openmm_tools.py`, and expanded `ToolSpec.inputs` to expose `simulation_steps`, `temperature_kelvin`, `friction_coeff_ps`, and `timestep_fs` as optional runtime parameters with their defaults.

-Replaced the water fixture in `test_openmm_tools.py` with a physically reasonable geometry and strengthened the tests to assert finite energies, energy reduction after minimization, correct zero-step behavior, and discoverability of the optional runtime parameters.

EmployeeNo427 committed:
-openmm_tools.py (line 94) fixes the nullable dict-key path that ty was complaining about.
-test_mcp_server_manager.py (line 22) removes the brittle hardcoded server count and checks the actual registered set, including mafft.
-ci.yml (line 217) pins ty to the repo baseline and scopes it to DeepResearch.
